proto/substrait/algebra.proto

```diff
@@ -726,6 +730,47 @@ message NestedLoopJoinRel {
   substrait.extensions.AdvancedExtension advanced_extension = 10;
 }
 
+message DelimGetRel{
+   RelCommon common = 1;
+   repeated  Type  delim_types = 2;
+}
+
+message DelimJoinRel {
+  RelCommon common = 1;
+  Rel left = 2;
+  Rel right = 3;
+
+  Expression expression = 4;
+  Expression post_join_filter = 5;
+
+  JoinType type = 6;
+
+  // The set of columns that will be duplicate eliminated from the LHS and pushed into the RHS
+  repeated Expression duplicate_eliminated_columns = 7;
+
+  // If this is a DelimJoin, whether it has been flipped to de-duplicating the RHS instead
```

Comment:
Is this flipping tied to the left/rightness of the join?  If it's always the same way then it could be determined by the left/rightness.  If it could be different, then the definition of what the normal behavior is would be useful so it's clear what flipped is.  Perhaps a deduplicate enum having left and right as options could be more readable?

Reply:
Yes, it basically tells us which side is being de-duplicated, I agree that an enum has a more clear intention, will change to that!